src/resources/elements/primeDesignSystem/pform-input/pform-input.ts

```diff
@@ -0,0 +1,53 @@
+import {BindingEngine, child, customElement} from "aurelia-framework";
+import "./pform-input.scss";
+import {bindable, observable} from "aurelia-typed-observable-plugin";
+import {ValidationState} from "../types";
+import {Disposable} from "aurelia-binding";
+
+@customElement("pform-input")
+export class PFormInput {
+  @bindable.string @observable validationState?: ValidationState;
+  @bindable.string label = "";
+  @bindable.string labelDescription = "";
+  @bindable validationMessage = "";
+  @bindable.boolean showCounter = false;
+  @bindable.number maxValue = 0;
+  @bindable.string helperMessage = "";
+  @child("*") input;
+
+  private inputValueObserverSubscription?: Disposable;
+
+  constructor(private element: Element, private bindingEngine: BindingEngine) {
+  }
+
+  attached() {
+    this.validationStateChanged(this.validationState);
+
+    if (this.input && this.showCounter) {
+      this.limitInputCharacterLength();
+    }
+  }
+
+  private limitInputCharacterLength() {
+    const inputValueObserver = this.bindingEngine.propertyObserver(this.input, "value");
+    this.inputValueObserverSubscription = inputValueObserver.subscribe(newValue => {
+      if (newValue?.length > this.maxValue) {
+        this.input.value = newValue.substring(0, this.maxValue);
+      }
+    });
+  }
+
+  validationStateChanged(newValue?: ValidationState) {
+    if (this.input) {
+      this.input.validationState = newValue;
+    }
+  }
+
+  validationStateExists(state: ValidationState){
```

Comment:
Nit: Need eslint-rule
Last characters here `){` (without a space)

Reply:
It's weird because I have this space on my local branch. And I don't have anything changed. Everything is pused.